torch/csrc/jit/script/compiler.cpp

```diff
@@ -394,13 +380,30 @@ static inline bool isIntUsedAsIntList(
          *arg.type == *ListType::ofInts() && arg.N;
 }
 
+template<class T>
+static Value* materializeConstant(T val, Graph& graph,
+    const SourceRange& r, std::unordered_map<T, Value*>& map) {
+  auto existing_constant = map.find(val);
+  if (existing_constant != map.end()) {
+    return existing_constant->second;
+  }
+
+  WithInsertPoint guard(graph.block()->nodes().front());
+  auto new_constant = graph.insertConstant(val, r);
+  map[val] = new_constant;
+
+  return new_constant;
+}
+
 at::optional<std::vector<Value*>> tryMatchSchema(
   const FunctionSchema& schema,
   const SourceRange& loc,
   Graph& graph,
   at::ArrayRef<NamedValue> inputs,
   at::ArrayRef<NamedValue> attributes,
-  std::ostream& failure_messages) {
+  std::ostream& failure_messages,
+  std::unordered_map<int64_t, Value*>* integral_constants,
```

Comment:
These are threaded through here, but then never used.

More generally, the constant table is not actually reaching all the places where constants might be inserted in compilation. 

I think the right balance, at least for now, is to use the tables inside the compiler struct where they are easy to access, but not to worry about them for functions that exist outside it (like jit/script/init.cpp functions or tryMatchSchema). These functions are exposed in other places, for instance I have a PR that uses tryMatchSchema so that passes can create schema-checked IR, and it would get awkard to add the additional context.

Reply:
It's an artifact from when I did tried to use materializeConstant around like 444, but it didn't work out and I forgot to undo it.